src/ProjectManager.cpp

```diff
@@ -124,62 +124,50 @@ void ProjectManager::SaveWindowSize()
       gPrefs->Write(wxT("/Window/Normal_X"), normalRect.GetX());
       gPrefs->Write(wxT("/Window/Normal_Y"), normalRect.GetY());
       gPrefs->Write(wxT("/Window/Normal_Width"), normalRect.GetWidth());
       gPrefs->Write(wxT("/Window/Normal_Height"), normalRect.GetHeight());
-      gPrefs->Write(wxT("/Window/Iconized"), FALSE);
+      gPrefs->Write(wxT("/Window/Iconized"), isIconized);
+      gPrefs->Flush();
+   };
+
+   if (validWindowForSaveWindowSize)
+   {
+      wxRect windowRect = validProject->GetRect();
+      wxRect normalRect = validProject->GetNormalizedWindowState();
+      bool wndMaximized = validProject->IsMaximized();
+
+      SaveWindowPreferences(windowRect, normalRect, wndMaximized, FALSE);
    }
    else
    {
       if (foundIconizedProject) {
          validProject = &ProjectWindow::Get( **AllProjects{}.begin() );
          bool wndMaximized = validProject->IsMaximized();
          wxRect normalRect = validProject->GetNormalizedWindowState();
+
          // store only the normal rectangle because the itemized rectangle
          // makes no sense for an opening project window
-         gPrefs->Write(wxT("/Window/X"), normalRect.GetX());
-         gPrefs->Write(wxT("/Window/Y"), normalRect.GetY());
-         gPrefs->Write(wxT("/Window/Width"), normalRect.GetWidth());
-         gPrefs->Write(wxT("/Window/Height"), normalRect.GetHeight());
-         gPrefs->Write(wxT("/Window/Maximized"), wndMaximized);
-         gPrefs->Write(wxT("/Window/Normal_X"), normalRect.GetX());
-         gPrefs->Write(wxT("/Window/Normal_Y"), normalRect.GetY());
-         gPrefs->Write(wxT("/Window/Normal_Width"), normalRect.GetWidth());
-         gPrefs->Write(wxT("/Window/Normal_Height"), normalRect.GetHeight());
-         gPrefs->Write(wxT("/Window/Iconized"), TRUE);
+         SaveWindowPreferences(normalRect, normalRect, wndMaximized, TRUE);
       }
       else {
          // this would be a very strange case that might possibly occur on the Mac
          // Audacity would have to be running with no projects open
          // in this case we are going to write only the default values
          wxRect defWndRect;
          GetDefaultWindowRect(&defWndRect);
-         gPrefs->Write(wxT("/Window/X"), defWndRect.GetX());
-         gPrefs->Write(wxT("/Window/Y"), defWndRect.GetY());
-         gPrefs->Write(wxT("/Window/Width"), defWndRect.GetWidth());
-         gPrefs->Write(wxT("/Window/Height"), defWndRect.GetHeight());
-         gPrefs->Write(wxT("/Window/Maximized"), FALSE);
-         gPrefs->Write(wxT("/Window/Normal_X"), defWndRect.GetX());
-         gPrefs->Write(wxT("/Window/Normal_Y"), defWndRect.GetY());
-         gPrefs->Write(wxT("/Window/Normal_Width"), defWndRect.GetWidth());
-         gPrefs->Write(wxT("/Window/Normal_Height"), defWndRect.GetHeight());
-         gPrefs->Write(wxT("/Window/Iconized"), FALSE);
+         SaveWindowPreferences(defWndRect, defWndRect, FALSE, FALSE);
```

Comment:
I like the elimination of duplication.

Let's be modern and use `true` and `false`.

Reply:
I have swapped out TRUE/FALSE for true/false, and removed duplications of the string literals "/Window/X", etc.